### DISCUSSION TITLE
Filter on numeric attribute and take care of absent value

### DISCUSSION THREAD

**Discussion by LeComptoirDesPharmacies** [algolia/instantsearch#5302] [Q&A] [answered]
Hi Everyone,

We are somewhat blocked with a capability that we want to provide to our customers.
It is a classic slider where user select a value (eg. 15) and we want to display all records where `stock` (aka. an attribute) is above.
This should be good using numeric filter like `stock >= 15`.

In addition, we have some records with an infinite number of stock so we create some records without `stock` attribute.
We followed this documentation [filter-by-null-or-missing-attributes](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/how-to/filter-by-null-or-missing-attributes/) and had the idea to put a tag `IS_LIMITED_STOCK` on records where `stock` is present. This would result in the following filter : `stock >= 15 OR NOT _tags:IS_LIMITED_STOCK`.

However there is one main problems : The following documentation [mixing-type-of-filters-in-or-conditions](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/in-depth/combining-boolean-operators/#mixing-type-of-filters-in-or-conditions) tell that we can not mix numeric and tags filters.

O.K ! So maybe use a custom attribute named "is_limited_stock" with a boolean value that we could use through numeric notation (See [Note about boolean attribute in numeric filters)](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/how-to/filter-by-boolean/#applying-a-boolean-filter).
This would result in the following filter : `stock >= 15 OR is_limited_stock = 0`.

It seems great and seems to work, however we do not succeed to get a proper way of include this filter using `react-instantsearch` library.
As we are using UI widget with the need to produce a refinement that can be cleared using [clearsquery](https://www.algolia.com/doc/api-reference/widgets/current-refinements/react/#widget-param-clearsquery) we do not want to use `Configure`.
Instead we created a copy of the `Range` connector and are ready to customize the `getSearchParameters` method to add the `OR is_limited_stock = 0` part.

But it seems easier writing than doing. 😫 
We found a ton of topic discussing this problem (See related issues below) but did not find a clear solution yet.

What are our best options ?
1. Use `setQueryParameter` on `filters` and break other `Configure` nodes of the application ?
Is there a way to get current `filters` value in `getSearchParameters` and maybe merge the new filter using 'and' ?
Could it avoid breaking `Configure` nodes ? Is it robust / production-ready ?
2. Always produce records with a `stock` attribute and, for unlimited one, put a magic infinite number like '99999' ?
This solution is easier in term of filtering but it will "corrupt" a bit the data and it force all my clients (app) to know that '99999' means unlimited and should be replaced/ignored.
3. Create a `stock` attribute sibling that will be filled like in point (2) but only used for filtering purpose.
Ex: `onlyFilterStock`
4. Any idea that you have in your bag will be welcomed. :pray: 

Related issues : 
https://github.com/algolia/algoliasearch-helper-js/issues/429
https://github.com/algolia/react-instantsearch/issues/686
https://github.com/algolia/algoliasearch-helper-js/issues/290
https://github.com/algolia/algoliasearch-helper-js/issues/748
https://github.com/algolia/react-instantsearch/issues/725

Yours faithfully,
LCDP

**Comment by Haroenv** [comment, marked as answer]
There's many ways to solve this, and ideally you would be able to put Infinity, but that doesn't work, but what do you think about adding a value of let's say 10000 (much higher than regular stock) so it shows up in the filter, and manually add a max value to the range slider?

**Comment by LeComptoirDesPharmacies** [comment]
Hi @Haroenv,

Thanks for this answer.  As we are using "min-only" filters, we followed your advice using solution (2).
In order for client to know if the value is a magical number, we added a boolean attribute next to it.

Yours faithfully,
LCDP

**Reply by Haroenv** [reply]
That sounds like a solid solution with these constraints!
